### ISSUE TITLE
Ability to sort apps in a custom way

### ISSUE THREAD

**Issue by meenbeese** [ImranR98/Obtainium#377]
Hello. Your app is really awesome, and perhaps even complete, but there is a feature that I could take a lot of advantages from. Currently, all apps are sorted alphabetically in the "Apps" section.

While this is enough for more users, the advanced users should have the option to change this. In order to not confuse anyone, a toggle could be added in the settings to explicitly turn this on. Then, the user could hold and reorder the listed apps.

Thank you for the great app again. Really appreciate it!

**Comment by S7venLights** [comment]
You can use the custom categories to easily filter apps by group…

**Comment by Rikj000** [comment]
> You can use the custom categories to easily filter apps by group…

That's true, but it would give a nice overview if all apps could be visible / not-filtered,   
while remaining grouped by category / color-code.

Additionally, I'd like to expand upon this feature request.
**Please make Categories sort-able too.**

Currently Categories are simply ordered in an auto-incremental way,   
with no ability to change the order (#377) / color (#397) afterwards.

This can be fairly annoying when you'd like to have a certain color for a category in the app overview.   
Because Obtainium will only display the color of the category which is selected + created the earliest.

## Issue Example
- Add 2 categories (BlueCat + RedCat),
    for simplicity, this gets worse with more categories, currently on my phone I have 9 categories.
- Keep on deleting + re-creating the categories until you reached the desired Blue + Red colors **(this is already an issue)**
- Apply 2 categories to an ExampleApp (BlueCat + RedCat)
- You'd like to change BlueCat now, and re-name/re-color it to GreenCat,   
   and also would like to keep the visibility of the GreenCat color in the app overview.

## Current flow issued
- Delete the BlueCat category
- Keep on deleting + re-creating the GreenCat category until you reached the desired Green color
- Apply GreenCat category to ExampleApp, conclude that the color in the app overview is now Red
- Keep on deleting + re-creating the RedCat category until you reached the desired Red color again,
   now it should be back below GreenCat
- Re-apply RedCat category to ExampleApp, conclude that the color in the app overview is now finally Green
- On my phone I'd have to also re-create 7 more categories in this tedious way..

## Desired flow / changes
Obtainium => Settings => Categories
- [ ] Tap BlueCat => PopUp
    - Text field to edit name
    - Color picker
- [ ] Hold + drag BlueCat => Change order
Obtainium => Apps => Filter
- [ ] CheckBox "Sort by category"
